test/client/autosuggest.spec.ts

```diff
@@ -0,0 +1,532 @@
+    const selected = generateAutosuggestSuggestion();
+    const transportArguments = {
+      method: 'post',
+      host: `${host.replace(/\/$/, '')}/${apiVersion}`,
+      url: '/autosuggest-selection',
+      query: { key: apiKey },
+      headers: { 'X-Api-Key': apiKey, ...HEADERS },
+      body: {
+        'raw-input': '',
+        selection: selected.words,
+        rank: selected.rank,
+        'source-api': 'text',
+      },
+    };
+
+    await client.onSelected(selected).should.resolvedWith(undefined);
+    transportSpy
+      .calledOnceWith(transportArguments)
+      .should.be.equal(true, 'transport arguments do not match');
+  });
+  context(
+    'should call /autosuggest-selection with initial request options override',
+    () => {
+      const input = `${CHANCE.word()}.${CHANCE.word()}.${CHANCE.letter()}`;
+      const nResults = CHANCE.natural();
+      const nFocusResults = CHANCE.natural();
+      const focus = generateCoordinate();
+      const clipToBoundingBox = {
+        southwest: { lat: 71.2, lng: -0.123 },
+        northeast: { lat: 91, lng: -0.12 },
+      };
+      const clipToCircle = {
+        center: generateCoordinate(),
+        radius: CHANCE.natural(),
+      };
+      const clipToCountry = [CHANCE.locale()];
+      const startEndCoordinate = generateCoordinate();
+      const clipToPolygon = [
+        startEndCoordinate,
+        generateCoordinate(),
+        generateCoordinate(),
+        generateCoordinate(),
+        startEndCoordinate,
+      ];
+      const language = CHANCE.locale();
+      const preferLand = CHANCE.bool();
+      const selected = generateAutosuggestSuggestion();
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      let transportArguments: { [key: string]: any };
+
+      beforeEach(() => {
+        transportArguments = {
+          method: 'post',
+          host: `${host.replace(/\/$/, '')}/${apiVersion}`,
+          url: '/autosuggest-selection',
+          query: { key: apiKey },
+          headers: { 'X-Api-Key': apiKey, ...HEADERS },
+          body: {
+            'raw-input': input,
+            selection: selected.words,
+            rank: selected.rank,
+            // 'source-api': 'text',
+            'n-results': `${nResults}`,
+            focus: `${focus.lat},${focus.lng}`,
+            'n-focus-results': `${nFocusResults}`,
+            'clip-to-bounding-box': `${clipToBoundingBox.southwest.lat},${clipToBoundingBox.southwest.lng},${clipToBoundingBox.northeast.lat},${clipToBoundingBox.northeast.lng}`,
+            'clip-to-circle': `${clipToCircle.center.lat},${clipToCircle.center.lng},${clipToCircle.radius}`,
+            'clip-to-country': clipToCountry.join(','),
+            'clip-to-polygon': clipToPolygon
+              .map(coord => `${coord.lat},${coord.lng}`)
+              .join(','),
+            language,
+            'prefer-land': `${preferLand}`,
+            // 'input-type': inputType,
```

Comment:
A couple of commented out lines here - are they meant to be here still?

Reply:
Good spot 👍